src/store/Disks.cc

```diff
@@ -386,6 +408,70 @@ Store::Disks::updateLimits()
     }
 }
 
+void
+Store::Disks::Parse(Store::DiskConfig &swap)
+{
+    const auto typeStr = ConfigParser::NextToken();
+    if (!typeStr)
+        throw TextException("missing cache_dir parameter: storage type", Here());
+
+    const auto pathStr = ConfigParser::NextToken();
+    if (!pathStr)
+        throw TextException("missing cache_dir parameter: directory name", Here());
+
+    const auto fs = StoreFileSystem::FindByType(typeStr);
+    if (!fs) {
+        debugs(3, DBG_PARSE_NOTE(DBG_IMPORTANT), "ERROR: This proxy does not support the '" << typeStr << "' cache type. Ignoring.");
+        return;
+    }
+
+    const auto fsType = fs->type();
+
+    // check for the existing cache_dir
```

Comment:
I would document another old out-of-scope XXX:

```suggestion
    // check for the existing cache_dir
    // XXX: This code mistreats duplicated cache_dir entries (that should be fatal).
```